meps/mep-cross-references.md

```diff
@@ -0,0 +1,339 @@
+
+```text
+URI = scheme ":" pathname ["?" query] "#" fragment
+```
+
+In most cases, as seen in the summary above the scheme is not required to be explicit and can be inferred safely by the context.
+The exception is when _explicitly_ referring to an external MyST site, Jupyter Book or Sphinx documentation site.
+These URIs can be safely and easily parsed by any common URL parser. For example in Javascript:
+
+```js
+const url = new URL('project:target.md#my-ref');
+url.protocol; // "project:"
+url.pathname; // "target.md"
+url.hash; // "#my-ref"
+```
+
+For most internally linked references, we expect the inline syntax to be most commonly used, with the autolink and scheme to only be used when specific behavior is intended. The following links and references are supported:
+
+| Link Type                | Auto Link                  | Inline                    |
+| :----------------------- | :------------------------- | :------------------------ |
+| External URL             | `<https://example.com>`    | `[](https://example.com)` |
+| Local file download      | `<path:file.txt>`          | `[](file.txt)`            |
+| File download (explicit) | `<path:file.md>`           | `[](path:file.md)`        |
+| Project document         | `<project:file.md>`        | `[](file.md)`             |
+| Target in a document     | `<project:target.md#file>` | `[](file.md#target)`      |
+| Target in project        | `<project:#target>`        | `[](#target)`             |
+
+### Search Order and Specificity
+
+All references search the local document first[^specific_doc], then the local project in the order of the table of contents. A `xref_multiple` warning is raised if multiple matches are found.
+
+[^specific_doc]: With the exception of an explicit reference to a specific page, i.e. `[](./examples/my-doc.md#explicit-reference)`
+
+In large documentation sites, a referenced target can be present in multiple documents, in that case, the parser will emit a `xref_multiple` warning letting you know that there are multiple matches for the intended target.
+
+### Implicit Section Headers
+
+We suggest a configuration option to create anchor "slugs" for section headers, which stay close to the [GitHub implementation](https://github.com/Flet/github-slugger), which produces references that:
+
+- lower-case text
+- remove punctuation
+- replace spaces with `-`
+- enforce uniqueness via suffix enumeration `-1`
+
+For example, `## Links and Referencing` can be referenced as `[](#links-and-referencing)`.
+These are **implicit** references, and referring to them should raise an `xref_implicit` warning, which can optionally be suppressed by users.
```

Comment:
Note, in myst-parser, I currently don't emit this warning, because it is optional anyway to turn on anchors, to a certain depth.

Is this MEP suggesting every heading level should have an anchor, I think that should be said explicitly

Also this MEP does not mention about a `[](path/to/file.md#heading-anchor)` format.
This is something that GitHub, VS Code, etc support, as does Myst-Parser. Should it be here?

Reply:
I have lightened the language, from `should` to `can`, I think that we will continue to raise that warning, so would like it documented, but also not a big deal as there are different ways to bring the configuration/parser together.

Added the text `Every heading level in a document should have an anchor`

The `[](path/to/file.md#heading-anchor)` is mentioned on around line 257 as "Target in a document", I didn't spend too long explaining it, but it is definitely there!